### ISSUE TITLE
Discussion article

### ISSUE THREAD

**Issue by mr-september** [PlusGenie/tbot-tradingboat#1]
Hi, not an issue, just a discussion. 

Have you seen this article? https://wasteofserver.com/interactive-brokers-tws-gateway-api-and-yet-it-works/

Not sure if anything in there could help improve this project?

**Comment by PlusGenie** [comment]
Hi Mr-September,

The TBOT project uses the ib_insync API instead of the TWS API.

In short, that's a kind of reason that I decided not to use the TWS API and then move to ib_insync. As I try to apply the software design pattern, I could see some redundancies coming from the TWS API's architecture.

For example, in the article you gave me, the author used Thread() to create a Queue(). One of my many questions was why I had to create Thread() separately? The code just sleeps and then lets someone wake it up so that I can make my robot extremely simple. More complicated, more intricately bugged.

This is a kind of reason that TBOT does not use Thread(), completely sleeping until it receives the message from Redis. Also, queuing is taken care of by Redis. Again, I didn't want to make the TBOT complicated. So someone should take care of queues. 

If you take a look at the source of the ib_insync API, you will realize that it completely removes the dependency on the TWS API.
